packages/react-core/src/components/Menu/Menu.tsx

```diff
@@ -0,0 +1,101 @@
+import * as React from 'react';
+import styles from '@patternfly/react-styles/css/components/Menu/menu';
+import { css } from '@patternfly/react-styles';
+import { getOUIAProps, OUIAProps, getDefaultOUIAId } from '../../helpers';
+import { SearchInput } from '../SearchInput';
+import { Divider } from '../Divider';
+import { MenuContext } from './MenuContext';
+
+export interface MenuProps extends Omit<React.HTMLAttributes<HTMLDivElement>, 'ref' | 'onSelect'>, OUIAProps {
+  /** Anything that can be rendered inside of the Menu */
+  children?: React.ReactNode;
+  /** Additional classes added to the Menu */
+  className?: string;
+  /** Callback for updating when item selection changes */
+  onSelect?: (event: React.MouseEvent, itemId: any) => void;
+  /** Callback called when an MenuItems's favorite button is clicked */
+  onFavorite?: (event: React.MouseEvent, itemId: any) => void;
+  /** A callback for when the input value changes. */
+  onSearchInputChange?: (
+    event: React.FormEvent<HTMLInputElement> | React.SyntheticEvent<HTMLButtonElement>,
+    value: string
+  ) => void;
+  /** Accessibility label */
+  'aria-label'?: string;
+  /** Indicates if menu is the flyout variant */
+  isFlyoutMenu?: boolean;
+  /** Search input of menu */
+  hasSearchInput?: boolean;
+  /** Forwarded ref */
+  innerRef?: React.Ref<any>;
+}
+
+export interface MenuState {
+  searchInputValue: string | null;
+  ouiaStateId: string;
+}
+
+class MenuBase extends React.Component<MenuProps, MenuState> {
+  static defaultProps: MenuProps = {
+    ouiaSafe: true
+  };
+
+  state: MenuState = {
+    ouiaStateId: getDefaultOUIAId(Menu.displayName),
+    searchInputValue: ''
+  };
+
+  render() {
+    const {
+      'aria-label': ariaLabel,
+      children,
+      className,
+      onSelect,
+      onFavorite,
+      onSearchInputChange,
+      ouiaId,
+      ouiaSafe,
+      isFlyoutMenu,
+      hasSearchInput,
+      innerRef,
+      ...props
+    } = this.props;
+
+    return (
+      <MenuContext.Provider value={{ onSelect, onFavorite }}>
+        <div
+          className={css(styles.menu, isFlyoutMenu && styles.modifiers.flyout, className)}
+          aria-label={ariaLabel || isFlyoutMenu ? 'Local' : 'Global'}
+          ref={innerRef}
+          {...getOUIAProps(Menu.displayName, ouiaId !== undefined ? ouiaId : this.state.ouiaStateId, ouiaSafe)}
+          {...props}
+        >
+          {hasSearchInput && (
+            <React.Fragment>
+              <div className={styles.menuSearch}>
+                <SearchInput
```

Comment:
@mcarrano Jenny is using the search input component as the search filter. Typically we build the search box in these menus out using a standard form control text input. Do you think this is the case for the search input component instead?

@jenny-s51 is there an advantage of using `<SearchInput>` over `<TextInput>`?

Reply:
@mcoker I think I used `<SearchInput>` here because it matched the styling in the core demo. The icon variant of `<TextInput>` has  the icon on the left as opposed to the right, but for the core demo of the filterable menu variant, the search icon was on the right so I figured `<SearchInput>` was the component needed here.

Reply:
@mcoker I don't have a preference for using one over the other. But it does seem like we should be consistent between core and React. If there is an advantage to using the Search input component I'm fine with that.

Reply:
@jenny-s51 ah! We support a variant of the `pf-c-form-control` component (used in `<TextInput>`) with `.pf-m-search` that creates the icon on the left. We use that in `<Select>` and `<TreeView>` so I tracked those down and looks like we're creating this for some reason:

`<input className={css(formStyles.formControl, formStyles.modifiers.search)} type="search" />`

https://github.com/patternfly/patternfly-react/blob/94bdfccf5cb96c1388fdf9528a145275264671e3/packages/react-core/src/components/TreeView/TreeViewSearch.tsx#L10

https://github.com/patternfly/patternfly-react/blob/94bdfccf5cb96c1388fdf9528a145275264671e3/packages/react-core/src/components/Select/Select.tsx#L709-L712

Is there any reason we shouldn't add this as a variant to `<TextInput>` and use that there instead? Is it because it might be confusing to have `type="search"` on `<TextInput>` that isn't necessarily tied to this search icon? Currently I think both `<input class="pf-m-form-control pf-m-search" type="text">` and `<input class="pf-m-form-control" type="search">` are valid - just meaning `pf-m-search` and `type="search"` don't have to be used together.

Though I agree with @mcarrano, we could use `<SearchInput>` if there is an advantage. WDYT @jenny-s51? I'll also add that to our meeting agenda for tomorrow @mcarrano.

Reply:
Using search input here made more sense to me intuitively, as the filterable variant uses searching functionality and we already had this component built out in react. I definitely agree that it's a bit confusing how currently `pf-m-search` and `type="search"` are used together on  `<input>` though ... I can open a new `TextInput` issue in react to address that @mcoker @mcarrano